### PR TITLE
feat: Adding `premium_progress_bar_enabled`

### PR DESCRIPTION
**Pull request by B1ue-Dev**
## About

Title. (I'm surprised that this is missing, somehow. Also, can someone do with `stickers` and `emojis` as well? My ability can't comprehend this lol.)

## Checklist

- [x] The ``pre-commit`` code linter has been run over all edited files to ensure the code is linted.
- [x] I've ensured the change(s) work on `3.8.6` and higher.
- [ ] I have added the `versionadded`, `versionchanged` and `deprecated` to any new or changed user-facing function I committed.

### Pull-Request specification

I've made this pull request: (check all that apply)
  - [ ] For the documentation
  - [x] To add a new feature
  - [ ] As a general enhancement
  - [ ] As a refactor of the library/the library's code
  - [ ] To fix an existing bug
  - [ ] To resolve #ISSUENUMBER

This is:
  - [ ] A breaking change

### CODE REVIEW

**Comment by B1ue-Dev** [comment]
Follow up the PR, I committed the group() fix as well.

File: interactions/models/internal/application_commands.py
Comment (LordOfPolls) [inline interactions/models/internal/application_commands.py:1]: This change needs to be made, however please can you separate these changes into separate prs

**Review by LordOfPolls — CHANGES_REQUESTED** [review]
